File: 프로그래머스/lv0/120861. 캐릭터의 좌표/캐릭터의 좌표.py
def solution(keyinput, board):
    x, y = [i // 2 for i in board];
    direction = {'up' : [0, 1],'down' : [0, -1],'left' : [-1, 0],'right' : [1, 0]};
    keyinput = [direction[dir] for dir in keyinput];
   
    point = [0, 0];
    for moveX, moveY in keyinput:
        point[0] = x if point[0] + moveX > x else x * -1 if point[0] + moveX < x * -1 else point[0] + moveX;
        point[1] = y if point[1] + moveY > y else y * -1 if point[1] + moveY < y * -1 else point[1] + moveY;
    
    return point
    

File: 프로그래머스/lv0/120861. 캐릭터의 좌표/캐릭터의 좌표.js
function solution(keyinput, board) {
    const [x, y] = board.map(el => Math.floor(el / 2));
    const direction = {'up': [0, 1], 'down':[0, -1], 'left':[-1, 0], 'right' : [1, 0]};
    
    return keyinput.reduce((acc, cur) => {
        let moveX = acc[0] + direction[cur][0] > x ? x : acc[0] + direction[cur][0] < (x * -1) ? x * -1 : acc[0] + direction[cur][0];
        let moveY = acc[1] + direction[cur][1] > y ? y : acc[1] + direction[cur][1] < (y * -1) ? y * -1 : acc[1] + direction[cur][1];
       
        return [moveX, moveY] 
    }, [0, 0])
}